tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc

```diff
@@ -677,15 +677,31 @@ TEST (nnstreamer_nnfw_mlapi, invoke_single_02_n)
   ml_tensors_info_destroy (in_res);
 }
 
-#define wait_for_sink(expected_cnt)                 \
-  do {                                              \
-    guint waiting_time = 0;                         \
-    while (*sink_called_cnt < expected_cnt) {       \
-      g_usleep (TEST_DEFAULT_SLEEP_TIME);           \
-      waiting_time += TEST_DEFAULT_SLEEP_TIME;      \
-      ASSERT_GT (TEST_TIMEOUT_LIMIT, waiting_time); \
-    }                                               \
-  } while (0);
+static GMutex g_test_mutex;
+
+/**
+ * @brief Wait until the sink gets the expected number of frames
+ */
+static gboolean
+wait_for_sink (guint expected_cnt)
+{
+  guint waiting_time = 0;
+  gboolean done = FALSE;
+
+  while (!done && waiting_time < TEST_TIMEOUT_LIMIT) {
+    g_mutex_lock (&g_test_mutex);
+
+    done = (*sink_called_cnt >= expected_cnt);
```

Comment:
Compile time error :)


```bash
[   43s] ../tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc:694:14: error: 'sink_called_cnt' was not declared in this scope

```

Reply:
That's right :) I updated the PR. Thanks.